docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
 
 This proposal aims to outline a design for cloud-edge communication framework/service to solve the above problems
 
 ## Goals
 
 * Support communication access capability of cloud-edge basic management components
-
 * Support independent IP addressing and IP address convergence of edge management components/services
-
 * Minimize attack surface, zero trust security
 
+## Proposal
+
+The following two solutions do not conflict with each other. The former can converge IP addresses to resolve IP address conflicts. However, the communication needs to be first sent to the gateway and then to the destination service. The latter cannot converge IP addresses but can resolve IP address conflicts. The latter provides end-to-end communication.
+
+### Proposal A
+
+The communication of cloud edge components/services is realized based on L4/L7 proxy, which is similar to kubedge's cloud-edge message channel.
+
+Cloudgateway and edgegateway are introduced into the central cloud and each edge site respectively. The edge site edgegateway supports connecting to the cloud gateway of the central cloud when it is started. The service communication between cloud and edge needs to go through these two gateways before arriving. Service access interfaces are provided on these two gateways to receive requests and route requests to target services.
+
+* Cloudgateway and edgegateway are installed and run in the central cloud and each edge site respectively
+* Edgegateway of Edge site can actively establish encrypted links with cloudgateway on the central side, and support two-way message transmission
+* Register the services of edge site and central cloud to cloudgateway
+* Edgegateway in the edge site acts as the service in the central cloud, and exposes the service access interface of the central cloud to the service in the edge site
+* Cloudgateway in the central cloud acts as the service in each edge site and exposes the service access interface of the edge site in the central cloud
+
+### Proposal B
+
+The access of cloud edge components/services is realized based on L3 layer network communication, but the network/IP is not exposed to the remote end, but the components/services are exposed to the remote end.
+
+![](Proposal-B1.png)
+
+Consider providing the concept of virtual presence, which is the local virtual presentation of remote services. In this way, virtual presence can be regarded as a local service with local IP addresses. These IP addresses have nothing to do with the network where the remote service is located, and can be allocated independently. In this way, the problem of unified planning of IP address of VPN solution in the whole network can be solved at the network level. From the perspective of security, virtual presence is completely isolated from the remote service. The client cannot directly access the network where the remote service is located. If the client is attacked, it can not directly attack the server.
+
+![](Proposal-B2.png)
```

Comment:
My understanding is: the services in cloud have internal IPs that can be accessed from cloud gateways, but not internet or edge sites. If this is the case, shouldn't "FW(ACL)" be put on the cloud service side in the digram?

Reply:
Yes. The FW is between the cloud service and cloud gateway. Traffic that needs to access cloud services at all edge sites must pass through the cloud gateway, pass through the FW, and finally reach the cloud service. The FW is used for traffic ACL from CloudGateway to CloudService.